packages/grafana-schema/src/schema/dashboard/v2alpha0/dashboard.schema.cue

```diff
@@ -0,0 +1,777 @@
+// should we make them conditional in the new schema as well? or should we make them required but default to false?
+
+// Variable types
+VariableValue: VariableValueSingle | [...VariableValueSingle]
+
+VariableValueSingle: string | bool | number | CustomVariableValue
+
+// Custom formatter variable
+CustomFormatterVariable: {
+  name: string
+  type: VariableType
+  multi: bool
+  includeAll: bool
+}
+
+// Custom variable value
+CustomVariableValue: {
+  // The format name or function used in the expression
+  formatter: *null | string | VariableCustomFormatterFn
+}
+
+// Custom formatter function
+VariableCustomFormatterFn: {
+  value: _
+  legacyVariableModel: {
+    name: string
+    type: VariableType
+    multi: bool
+    includeAll: bool
+  }
+  legacyDefaultFormatter?: VariableCustomFormatterFn
+}
+
+// Dashboard variable type
+		// `query`: Query-generated list of values such as metric names, server names, sensor IDs, data centers, and so on.
+		// `adhoc`: Key/value filters that are automatically added to all metric queries for a data source (Prometheus, Loki, InfluxDB, and Elasticsearch only).
+		// `constant`: 	Define a hidden constant.
+		// `datasource`: Quickly change the data source for an entire dashboard.
+		// `interval`: Interval variables represent time spans.
+		// `textbox`: Display a free text input field with an optional default value.
+		// `custom`: Define the variable options manually using a comma-separated list.
+		// `system`: Variables defined by Grafana. See: https://grafana.com/docs/grafana/latest/dashboards/variables/add-template-variables/#global-variables
+VariableType: "query" | "adhoc" | "groupby" | "constant" | "datasource" | "interval" | "textbox" | "custom" |
+			"system" | "snapshot" @cog(kind="type")
+
+// Sort variable options
+// Accepted values are:
+// `0`: No sorting
+// `1`: Alphabetical ASC
+// `2`: Alphabetical DESC
+// `3`: Numerical ASC
+// `4`: Numerical DESC
+// `5`: Alphabetical Case Insensitive ASC
+// `6`: Alphabetical Case Insensitive DESC
+// `7`: Natural ASC
+// `8`: Natural DESC
+// VariableSort enum with default value
+// FIXME: I can't mark 0 as default
+VariableSort: 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 @cog(kind="enum",memberNames="disabled|alphabeticalAsc|alphabeticalDesc|numericalAsc|numericalDesc|alphabeticalCaseInsensitiveAsc|alphabeticalCaseInsensitiveDesc|naturalAsc|naturalDesc")
+
+// Options to config when to refresh a variable
+// `0`: Never refresh the variable
+// `1`: Queries the data source every time the dashboard loads.
+// `2`: Queries the data source when the dashboard time range changes.
+VariableRefresh: 0 | 1 | 2 @cog(kind="enum",memberNames="never|onDashboardLoad|onTimeRangeChanged")
+
+// Determine if the variable shows on dashboard
+// Accepted values are 0 (show label and value), 1 (show value only), 2 (show nothing).
+VariableHide: 0 | 1 | 2 @cog(kind="enum",memberNames="dontHide|hideLabel|hideVariable")
```

Comment:
Avoid numbers and use enums

Reply:
Yes. I personally struggle with numeric values like this. When reading a json you literally don't what what the number represent. With the v2 schema we have a good chance to change it, so I suggest let's expose those as string enums and handle the mapping to runtime-supporte values  in the transformer.

Reply:
Done! 100%!